### PR TITLE
Receive multiple laserscan

### PR DESCRIPTION
**Pull request by dom-lee**
Graph Navigation can receive multiple laserscan data.
change `laser_topics` in `config/navigation.lua` file

### CODE REVIEW

File: src/navigation/navigation_main.cc [hunk 1]
```diff
@@ -180,44 +189,51 @@ void OdometryCallback(const nav_msgs::Odometry& msg) {
...
   }
-  CHECK_EQ(cached_rays_.size(), msg.ranges.size());
-  point_cloud_.resize(cached_rays_.size());
-  for (size_t i = 0; i < cached_num_rays_; ++i) {
+  CHECK_EQ(cache.rays.size(), msg.ranges.size());
+
+  size_t start_idx = point_cloud_.size();
+  point_cloud_.resize(start_idx + cache.rays.size());
+  // Location of the laser on the robot. Assumes the laser is forward-facing.
+  const Vector2f kLaserLoc(CONFIG_laser_loc_x, CONFIG_laser_loc_y);
```
Comment (joydeep-b) [inline src/navigation/navigation_main.cc]: This won't work for multiple topics. Let's print a warning if the offset is non-zero, and multiple laser topics are subscribed.

**Review by joydeep-b — CHANGES_REQUESTED** [review]
One small change needed: print warning when using laser offset with multiple topics.

File: src/navigation/navigation_main.cc (hunk 1 above)
Comment (dom-lee) [reply src/navigation/navigation_main.cc]: Should we remove the offset and verify if the `frame_id` of messages from multiple topics is set to `base_link` (defined in a configuration file), issuing a warning if so?"

Comment (dom-lee) [reply src/navigation/navigation_main.cc]: I push the changes

**Review by dom-lee — COMMENTED** [review]
remove the offset and verify if the frame_id of messages from multiple topics is set to the `laser_frame` defined in a configuration file

**Comment by joydeep-b** [comment]
@dom-lee Removing the offset entirely will break functionality on some robots like the cars and the CoBot that have single LIDAR. You could convert to a list of offsets, or read from the TF tree - but one or the other is needed.

File: src/navigation/navigation_main.cc [hunk 2]
```diff
@@ -188,6 +190,31 @@ void OdometryCallback(const nav_msgs::Odometry& msg) {
   navigation_.UpdateOdometry(odom_);
 }
 
+void RetreveTransform(const std_msgs::Header& msg,
```
Comment (joydeep-b) [inline src/navigation/navigation_main.cc]: `RetrieveTransform` : Spelling

**Review by joydeep-b — CHANGES_REQUESTED** [review]

File: src/navigation/navigation_main.cc (hunk 2 above)
Comment (dom-lee) [reply src/navigation/navigation_main.cc]: Modified

**Review by joydeep-b — APPROVED** [review]

**Review by joydeep-b — APPROVED** [review]

**Comment by dom-lee** [comment]
fixed transformation order
`point_cloud_[start_idx + i] = (frame_tf * (r * cache.rays[i])).head<2>();`

**Pull request merged**